internal/transport/handler_server.go

```diff
@@ -361,7 +362,9 @@ func (ht *serverHandlerTransport) HandleStreams(startStream func(*Stream), trace
 		for buf := make([]byte, readSize); ; {
 			n, err := req.Body.Read(buf)
 			if n > 0 {
-				s.buf.put(recvMsg{data: buf[:n:n]})
+				buffer := new(bytes.Buffer)
+				buffer.Write(buf[:n])
```

Comment:
Did you try: `buffer := bytes.NewBuffer(buf[:n:n])`?  Would that avoid a copy and an alloc?